### PR TITLE
Adding podcast_ownership controller and pundit policy

### PR DESCRIPTION
**Pull request by Manasa2850**
What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Adding podcast_ownership controller and an associated pundit policy

## Related Tickets & Documents
This is an extension to https://github.com/forem/forem/pull/11410
Frontend / view for this controller will be worked on separately

## Added tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [x] No documentation needed

## [optional] Are there any post deployment tasks we need to perform?
No

### CODE REVIEW

**Comment by Manasa2850** [comment]
@rhymes could you please review this and let me know if the controller and pundit policy is fine? I can then start writing tests for them.

**Comment by rhymes** [comment]
@Manasa2850 sure, they seem fine, I'll do a formal review when tests are added. Thank you!

**Comment by Manasa2850** [comment]
@rhymes I get an error with this statement `let(:user)       { build_stubbed(:owner) }` in the pundit policy tests. 
Could you please guide me regarding this?

**Comment by rhymes** [comment]
@Manasa2850 `build`, `build_stubbed`, `create` factories require the name of the factory, see the `spec/factories` folder that have been registered. `owner` is the name of the virtual relation between a podcast and its user's owner, but what you need is to fabricate a `:user`

**Comment by Manasa2850** [comment]
@rhymes could you please review this? I've added the tests

**Comment by rhymes** [comment]
@Manasa2850 you still have a failing test, please investigate. This is also missing request specs for controllers, see for example: https://github.com/forem/forem/tree/master/spec/requests

**Comment by Manasa2850** [comment]
@rhymes I had a doubt regarding the controller specs.
When I created the controller using `rails generate`, a default `podcast_ownerships_request_spec.rb` file was created. Should I write the specs here or create separate files for `create, destroy and update` functions in the `spec/requests/podcast` folder?

**Comment by rhymes** [comment]
@Manasa2850 I think having `spec/requests/podcasts/podcast_ownerships_destroy_spec.rb` with all the specs for the various methods is best

**Comment by Manasa2850** [comment]
@rhymes I'm going to have 3 files
`spec/requests/podcasts/podcast_ownerships_destroy_spec.rb`
`spec/requests/podcasts/podcast_ownerships_create_spec.rb`
`spec/requests/podcasts/podcast_ownerships_update_spec.rb`

**Comment by rhymes** [comment]
@Manasa2850 I think one file is good enough but as you wish :)

**Comment by Manasa2850** [comment]
@rhymes I just realized that we don't really have a use for the `update` method :sweat_smile: 
The only thing required is to add or delete a user as the owner of a podcast which can be achieved by the `create` and `destroy` methods.
What do you think about this?

**Comment by rhymes** [comment]
@Manasa2850 I answered you here https://github.com/MLH-Fellowship/dev.to/issues/81#issuecomment-735215087

File: app/controllers/podcast_ownerships_controller.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  end
+
+  def create
+    @podcast_ownership = PodcastOwnership.new(podcast_ownership_params)
+
+    if @podcast_ownership.save
+      flash[:success] = "Owner created successfully"
+      redirect_to pod_path
+    else
+      flash[:error] = "User is already an owner"
```
Comment (mkcode) [inline app/controllers/podcast_ownerships_controller.rb:15]: Use error object on podcast ownership. See: https://guides.rubyonrails.org/active_record_validations.html#valid-questionmark-and-invalid-questionmark

File: app/controllers/admin/podcast_ownerships_controller.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    private
+
+    def podcast_ownership_params
+      allowed_params = %i[
+        podcast_id
+        user_id
+      ]
+      params.require(:podcast_ownership).permit(allowed_params)
+    end
+  end
```
Comment (mkcode) [inline app/controllers/admin/podcast_ownerships_controller.rb:14]: Move this to PodcastOwnershipsController where it is being called.

File: app/controllers/podcast_ownerships_controller.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    if @podcast_ownership.update(podcast_ownership_params)
+      @podcast_ownership.save
+      flash[:success] = "Updated successfully"
+      redirect_to pod_path
+    else
+      flash[:error] = "Error Updating!"
+      format.html { render :edit }
+    end
+  end
```
Comment (mkcode) [inline app/controllers/podcast_ownerships_controller.rb:35]: Nothing to update. Just create and destory.

File: app/controllers/podcast_ownerships_controller.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  before_action :authenticate_user!
+
+  def new
+    @podcast_ownership = PodcastOwnership.new
+  end
+
+  def create
+    @podcast_ownership = PodcastOwnership.new(podcast_ownership_params)
+
+    if @podcast_ownership.save
```
Comment (mkcode) [inline app/controllers/podcast_ownerships_controller.rb:11]: Let's use the policy in the controller.

Comment (mkcode) [reply app/controllers/podcast_ownerships_controller.rb:11]: Via the `authorize` method.

**Review by mkcode — CHANGES_REQUESTED** [review]
Address comments

**Comment by msarit** [comment]
Hello @Manasa2850! 👋🏾 

Drawing your attention to this PR; looks like you have some feedback to incorporate.

Let me know if you have any questions or need help. 🙏🏾

**Comment by Manasa2850** [comment]
Hi @msarit. I was working on this feature as part of the MLH Fellowship. I'll not be able to continue working on this anytime soon. Please feel free to close this PR. 
Thank you!

**Pull request closed without merging**